kedro-org/kedro-viz
# Kedro Viz Lite User Warning UI

## Description

Resolves #2058 

## Development notes

**Backend changes:**
* Refactored PackageCompatibilityAPI response to MetadataAPIResponse
* Updated the API endpoint from [GET] "/package-compatibilities" to "/metadata" and updated the response model
* Moved`get_package_compatibilities` to utils
* Created models to hold `PackageCompatibility` and `Metadata` for the application

**Frontend changes:**
* Updated function calls to fetchMetadata as per the backend updated endpoint
* Added `showBanner` to the redux store and created generic actions, reducers to extend banner functionality in future
* Created Banner components and added styles
* Added tests for banner component

Thank you @jitu5 for helping out in resolving the redux states 💯 

<img width="2045" alt="image" src="https://github.com/user-attachments/assets/d103c794-560c-423f-9382-1fb6a1e5a44a"><br>



<img width="2047" alt="image" src="https://github.com/user-attachments/assets/65bc55b6-fec7-421e-8057-ba7b8a6c33cd">


## QA notes

* All tests should pass
* You should see the banner when there are missing dependencies in your python env and you do `kedro viz --lite`. The banner should not appear again for the rest of the browser session once you close it.
* The `Learn More` button click should take you to the docs for `--lite`
* There should not be any changes with the `package_compatibility` feature we have in deploy and ET.

## Checklist
